mm2src/mm2_metrics/src/native.rs

```diff
@@ -0,0 +1,491 @@
+use common::log::error;
+use common::{executor::{spawn, Timer},
+             log::{LogArc, LogWeak}};
+use gstuff::ERRL;
+use metrics::{IntoLabels, Key, KeyName, Label};
+use metrics_core::ScopedString;
+use metrics_exporter_prometheus::PrometheusBuilder;
+use serde_json::Value;
+use std::{collections::HashMap,
+          fmt::Display,
+          io::Sink,
+          sync::{atomic::Ordering, Arc}};
+
+use crate::{common::log::Tag,
+            recorder::{labels_to_tags, name_value_map_to_message},
+            ClockOps, MetricsOps, MmRecorder};
+
+const QUANTILES: &[f64] = &[0.0, 1.0];
+
+pub type MetricName = ScopedString;
+
+type MetricLabels = Vec<Label>;
+
+pub type MetricNameValueMap = HashMap<String, PreparedMetric>;
+
+/// Increment counter if an MmArc is not dropped yet and metrics system is initialized already.
+#[macro_export]
+macro_rules! mm_counter {
+    ($metrics:expr, $name:expr, $value:expr) => {{
+        use $crate::metrics::Recorder;
+        if let Some(recorder) = $crate::TryRecorder::try_recorder(&$metrics){
+            let key = key_from_str($name);
+            let counter = recorder.register_counter(&key);
+            counter.increment($value);
+        };
+    }};
+
+    // Register and increment counter with label
+    ($metrics:expr, $name:expr, $value:expr, $($label_key:expr => $label_val:expr),+) => {{
+        use $crate::metrics::Recorder;
+        use $crate::native::from_slice_to_labels;
+        if let Some(recorder) = $crate::TryRecorder::try_recorder(&$metrics){
+            let key = $crate::metrics::Key::from_parts($name, from_slice_to_labels(vec![$($label_key.to_owned(), $label_val.to_owned()),+]));
+            let counter = recorder.register_counter(&key);
+            counter.increment($value);
+        };
+    }};
+}
+
+/// Update gauge if an MmArc is not dropped yet and metrics system is initialized already.
+#[macro_export]
+macro_rules! mm_gauge {
```

Comment:
If you remove all uses of the `mm_gauge` macro, we can leave this macro anyway as it can be used later.